### PR TITLE
Add provider locations filters to provider interface

### PR DESCRIPTION
**Pull request by willmcb**
## Context

Allows users to filter applications by provider's locations on the provider user interface.

## Changes proposed in this pull request

**Before**
Screenshot 2020-05-22 at 14 52 48

**After**
Screenshot 2020-05-22 at 14 51 55

## Guidance to review

- are the tests specific enough?
- to the names make sense to you?
- can you find any bugs in the way that the filter operates, try different combinations etc.
- location filters shouldn't show on the page before a provider is filtered for.
- filters for locations should only appear if their provider organisation is filtered for.

## Link to Trello card

https://trello.com/c/kISeiFeX/2099-build-add-training-location-attribute-to-application-view-list-filter

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/provider_interface/provider_options_service.rb [hunk 1]
```diff
@@ -30,6 +30,21 @@ def providers
         .distinct
     end
 
+    # TODO: Write test from this in provider_options_service_spec
```
Comment (duncanjbrown) [inline app/services/provider_interface/provider_options_service.rb:33]: Looks good — please add the spec!

File: spec/components/previews/provider_interface/filter_component_preview.rb [hunk 2]
```diff
@@ -1,9 +1,13 @@
 module ProviderInterface
   class FilterComponentPreview < ViewComponent::Preview
-    def fully_selected
+    def defualt_view
```
Comment (duncanjbrown) [inline spec/components/previews/provider_interface/filter_component_preview.rb:3]: typo: `defualt` for `default`

File: spec/components/previews/provider_interface/filter_component_preview.rb [hunk 3]
```diff
@@ -39,5 +43,34 @@ def with_name_status_provider_and_accredited_provider
          name: 'accredited_provider',
          options: [{ value: 5, label: 'Coventry University', checked: nil }] }]
     end
+
+    def with_name_status_provider_accredited_provider_and_locations
+      [{ type: :search, heading: 'Candidate’s name', value: '', name: 'candidate_name' },
```
Comment (duncanjbrown) [inline spec/components/previews/provider_interface/filter_component_preview.rb:48]: Love to see the data all laid out!

File: spec/system/provider_interface/provider_applications_filter_spec.rb [hunk 4]
```diff
@@ -47,11 +56,56 @@
...
     then_i_expect_the_relevant_accredited_provider_tags_to_be_visible
-
     when_i_click_to_remove_an_accredited_provider_tag
+
+    when_i_filter_by_providers
+    then_i_should_see_locations_that_belong_to_all_of_the_selected_providers_that_have_more_than_one_site
+
+    when_i_clear_the_filters
+    when_i_filter_by_a_specific_provider
+    then_i_should_only_see_locations_that_belog_to_that_provider
```
Comment (duncanjbrown) [inline spec/system/provider_interface/provider_applications_filter_spec.rb:66]: Typo, `belog` for `belong`

File: spec/models/provider_interface/provider_applications_page_state_spec.rb [hunk 5]
```diff
@@ -37,6 +40,26 @@
...
+    it 'can return filter config for a list of provider locations' do
+      page_state = described_class.new(params: ActionController::Parameters.new({ provider: [provider1.id] }),
+
+                                       provider_user: another_provider_user)
+
+      headings = page_state.filters.map { |filter| filter[:heading] }
+
+      expect(headings).to include("Locations for #{provider1.name}")
+
+      # TODO: think if more specific test that tests multiple provider_locations
```
Comment (duncanjbrown) [inline spec/models/provider_interface/provider_applications_page_state_spec.rb:53]: I think it's ok not to get too into the nitty-gritty of this object — it's tested pretty thoroughly through the view

**Comment by duncanjbrown** [comment]
Minor comments inline! I see a blank "Accredited provider" section on the review app?

Screenshot 2020-05-27 at 10 56 25

File: app/services/provider_interface/provider_options_service.rb (hunk 1 above)
Comment (willmcb) [reply app/services/provider_interface/provider_options_service.rb]: Done 👍

Comment (willmcb) [reply app/services/provider_interface/provider_options_service.rb]: Done 👍

**Pull request merged**